### ISSUE TITLE
Formating and code action not working as expected (elixir)

### ISSUE THREAD

**Issue by jalberto** [jose-elias-alvarez/null-ls.nvim#84]
### FAQ

- [X] I have checked [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

v0.5.0

### Steps to reproduce

- set lsp-config
- set null-ls with `require("null-ls").builtins.formatting.mix`
- set lightbulb and telescope
- write some incorrect code
- display code actions (no action to format)
- run `vim.lsp.buf.formatting()`
- prompt : (1) elixir, (2) null-ls
- if i select 2 nothing happens
- if I select 1 formating happens

### Expected behavior

Display code actions for formatting and make null-ls formatting works

### Actual behavior

- if i select 2 nothing happens
- if I select 1 formating happens

### Debug log

Cannot see anything relevant

### Help

No

**Comment by jalberto** [comment]
```
received LSP formatting request
running generators for method NULL_LS_FORMATTING
spawning command mix with args:
{ "format", "-" }
```

**Comment by jose-elias-alvarez** [comment]
A little bit confused about this one, since LSP formatting won't show up under code actions (both for actual LSPs and null-ls). However, it should work when you run `vim.lsp.buf.formatting()` and select null-ls, so something else is going on. To verify, do you have `mix` available on your path, and does `mix format` work as expected from the command line? 

I just tried it out on an Elixir file from a sample repo and it seems to work fine, so it could be a config issue. Could you post your full null-ls config?

**Comment by systemctl603** [comment]
I also have the issue.

I can confirm that running `:lua vim.lsp.buf.formatting()` in the menu does not format. However, using the default language server which has formatting capabilities, the formatting works.

Running the formatter from the path also works, and formats the file correctly.

Configuration: https://github.com/systemctl603/neovim-config

**Comment by jose-elias-alvarez** [comment]
@systemctl603 Can you post your debug output? Your config seems fine. Also, can you verify that you see null-ls when you run `:LspInfo`?

**Comment by systemctl603** [comment]
```
running generators for method NULL_LS_DIAGNOSTICS
running generators for method NULL_LS_DIAGNOSTICS
running generators for method NULL_LS_DIAGNOSTICS
received LSP formatting request
running generators for method NULL_LS_FORMATTING
spawning command prettier with args:
{ "--stdin-filepath", "$FILENAME" }
error output: nil
output: nil
received edits from generators
{}
successfully applied edits
```

![image](https://user-images.githubusercontent.com/59405008/128635123-2c7ebc24-a2a3-4e18-8f3b-daf8d2bc0b30.png)

**Comment by jose-elias-alvarez** [comment]
Thanks! I see that you're on Windows, and so I suspect this is related to #95. @jalberto Are you also on Windows? 

As I said there, I'm going to see if I can get some help on this from the Neovim Matrix chat, but since I don't know anything about Windows and don't have access to a Windows setup to test this on, there's not much I can do without a Windows user who's familiar with `uv.spawn`.

**Comment by jose-elias-alvarez** [comment]
Not sure if this is the same issue that was closed by 07fd5abcab09a370f8d15fc437f79becb121e025, but could you try updating to the latest version and see if it makes a difference?

**Comment by jalberto** [comment]
Sorry for the late reply.
I am on Linux and if I run the forma tin CLI it works as expected.
Also, I noticed something, 1st time I select null-ls do nothing, but if I repeat the command and select again null-ls, then it works as expected.

**Comment by jose-elias-alvarez** [comment]
Thanks for confirming. I'm testing this out a bit more and I can see that `mix format` has some unusual behavior. Could you quickly verify whether another formatter (e.g. Prettier) is working as expected?

**Comment by jalberto** [comment]
Yes, prettier seems to be working as expected (at least on html)

**Comment by jose-elias-alvarez** [comment]
Got it. I can see that `mix format` has a weird interaction with the way null-ls spawns processes and gets their output, and I have to determine whether it's a bug in my implementation, Neovim / luv, or the formatter itself, so this might take a bit.

**Comment by jose-elias-alvarez** [comment]
@jalberto When you get a chance, could you please try [this branch](https://github.com/jose-elias-alvarez/null-ls.nvim/tree/fix-spawn-stdout) of the plugin and see if mix works for you?

**Comment by jalberto** [comment]
`✗ Failed to update jose-elias-alvarez/null-ls.nvim/fix-spawn-stdout`

Seem Packer doesn't like that branch

**Comment by jose-elias-alvarez** [comment]
For Packer I think you need to set the branch like this:

```lua
use({ "jose-elias-alvarez/null-ls.nvim/", branch = "fix-spawn-stdout" })
```

You might also need to comment out the plugin, uninstall it, and then re-install it to get the new branch to work (though I think Packer might have fixed this).

**Comment by jalberto** [comment]
Removing/adding did the trick!

It seems to be working now at 1st trigger

Thanks!

**Comment by jose-elias-alvarez** [comment]
Great! I need to review and merge another PR first, but once I do, I'll merge the `fix-spawn-stdout` branch and close this issue.

**Comment by jalberto** [comment]
Thanks for the fixing and your time :)

**Comment by jose-elias-alvarez** [comment]
Fix merged in b4a11d8cd71ca0d7f7eb2b360f497194eed061f2.
